src/slurm_plugin/fleet_manager.py

```diff
@@ -53,11 +54,12 @@ def __hash__(self):
     @staticmethod
     def from_describe_instance_data(instance_info):
         try:
-            private_ip, private_dns_name = get_private_ip_address_and_dns_name(instance_info)
+            private_ip, private_dns_name, all_private_ips = get_private_ip_address_and_dns_name(instance_info)
             return EC2Instance(
                 instance_info["InstanceId"],
                 private_ip,
                 private_dns_name.split(".")[0],
```

Comment:
Have you tested the case when `SlurmSettings/Dns/UseEc2Hostnames` is set to `True`.
In this case I would expect the possibility for the DNS name to not match with the private IP. Does this condition generates problem?

Reply:
No, I haven't. I am running the test now. 
But IMO, the mismatch should now happen, because you fixed it by https://github.com/aws/aws-parallelcluster-node/pull/556

Reply:
I ran test_multiple_nics with additional cluster config:
```
  SlurmSettings:
    Dns:
      DisableManagedDns: true
      UseEc2Hostnames: true
```
The test passed